spinalcordtoolbox/moco.py

```diff
@@ -0,0 +1,894 @@
+                    cmd += ['--masks', im_mask.absolutepath]
+                else:
+                    # Mask only contains zeros. Copying the image instead of estimating registration.
+                    sct.copy(file_src, file_out_concat, verbose=0)
+                    do_registration = False
+                    # TODO: create affine mat file with identity, in case used by -g 2
+        # 3D mode
+        else:
+            cmd = ['isct_antsSliceRegularizedRegistration',
+                   '--polydegree', param.poly,
+                   '--transform', 'Translation[%s]' %param.gradStep,
+                   '--metric', param.metric + '[' + file_dest + ',' + file_src + ',1,' + metric_radius + ',Regular,' + param.sampling + ']',
+                   '--iterations', param.iter,
+                   '--shrinkFactors', '1',
+                   '--smoothingSigmas', param.smooth,
+                   '--verbose', '1',
+                   '--output', '[' + file_mat + ',' + file_out_concat + ']']
+            cmd += sct.get_interpolation('isct_antsSliceRegularizedRegistration', param.interp)
+            if im_mask is not None:
+                cmd += ['--mask', im_mask.absolutepath]
+        # run command
+        if do_registration:
+            kw.update(dict(is_sct_binary=True))
+            env = dict()
+            env.update(os.environ)
+            env = kw.get("env", env)
+            # reducing the number of CPU used for moco (see issue #201)
+            env["ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS"] = "1"
+            status, output = sct.run(cmd, verbose=1 if param.verbose == 2 else 0, **kw)
+
+    elif param.todo == 'apply':
+        sct_apply_transfo.main(args=['-i', file_src,
+                                     '-d', file_dest,
+                                     '-w', file_mat + param.suffix_mat,
+                                     '-o', file_out_concat,
+                                     '-x', param.interp,
+                                     '-v', '0'])
+
+    # check if output file exists
+    if not os.path.isfile(file_out_concat):
+        # sct.printv(output, verbose, 'error')
+        sct.printv('WARNING in ' + os.path.basename(__file__) + ': No output. Maybe related to improper calculation of '
+                                                                'mutual information. Either the mask you provided is '
+                                                                'too small, or the subject moved a lot. If you see too '
+                                                                'many messages like this try with a bigger mask. '
+                                                                'Using previous transformation for this volume (if it'
+                                                                'exists).', param.verbose, 'warning')
+        failed_transfo = 1
+
+    # TODO: if sagittal, copy header (because ANTs screws it) and add singleton in 3rd dimension (for z-concatenation)
```

Comment:
It looks like this comment describes what's happening here, so it shouldn't be 'TODO'

Reply:
oopsi-- fixed 5646e5fda58a8f816413c37cc392a441dca6d0b8